### PR TITLE
Feat/student guardian retrieve by id service

### PR DESCRIPTION
**Pull request by AgentEnder**

### CODE REVIEW

**Comment by hassanhabib** [comment]
@AgentEnder please address conflicts.
@eriadhami please review this PR.

**Comment by AgentEnder** [comment]
@hassanhabib @eriadhami conflicts resolved

File: OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs [hunk 1]
```diff
@@ -88,5 +88,46 @@ public async Task ShouldThrowValidatonExceptionOnRetrieveWhenGuardianIdIsInvalid
...
             this.dateTimeBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRetrieveWhenStorageStudentGuardianIsInvalidAndLogItAsync()
+        {
+            // given
+            DateTimeOffset randomDateTime = GetRandomDateTime();
+            StudentGuardian randomStudentGuardian = CreateRandomStudentGuardian(randomDateTime);
+            Guid inputSemasterCourseId = randomStudentGuardian.GuardianId;
```
Comment (eriadhami) [inline OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs:95]: @AgentEnder maybe it is better to have `inputGuardianId` here

File: OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs [hunk 2]
```diff
@@ -88,5 +88,46 @@ public async Task ShouldThrowValidatonExceptionOnRetrieveWhenGuardianIdIsInvalid
...
+            Guid inputStudentId = randomStudentGuardian.StudentId;
+            StudentGuardian nullStorageStudentGuardian = null;
+
+            var notFoundStudentGuardianException =
+                new NotFoundStudentGuardianException(inputStudentId, inputSemasterCourseId);
```
Comment (eriadhami) [inline OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs:100]: @AgentEnder same here

File: OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs [hunk 3]
```diff
@@ -88,5 +88,46 @@ public async Task ShouldThrowValidatonExceptionOnRetrieveWhenGuardianIdIsInvalid
...
+
+            var expectedSemesterCourseValidationException =
+                new StudentGuardianValidationException(notFoundStudentGuardianException);
+
+            this.storageBrokerMock.Setup(broker =>
+                 broker.SelectStudentGuardianByIdAsync(inputStudentId, inputSemasterCourseId))
```
Comment (eriadhami) [inline OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs:106]: @AgentEnder here too

File: OtripleS.Web.Api/Services/StudentGuardians/StudentGuardianService.cs [hunk 4]
```diff
@@ -61,10 +61,12 @@ public partial class StudentGuardianService : IStudentGuardianService
 		});
 
 		public ValueTask<StudentGuardian> RetrieveStudentGuardianByIdAsync(Guid studentId, Guid guardianId) =>
-		TryCatch(() =>
+		TryCatch(async () =>
 		{
 			ValidateStudentGuardianIdIsNull(studentId, guardianId);
-			return this.storageBroker.SelectStudentGuardianByIdAsync(studentId, guardianId);
+			StudentGuardian storageStudentGuardian = await this.storageBroker.SelectStudentGuardianByIdAsync(studentId, guardianId);
```
Comment (eriadhami) [inline OtripleS.Web.Api/Services/StudentGuardians/StudentGuardianService.cs]: @AgentEnder look like too many characters in one line

File: OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs (hunk 1 above)
Comment (AgentEnder) [reply OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs]: Ah, my bad. I used the StudentSemesterCourse tests as a template and missed this due to Semaster vs Semester.

File: OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs (hunk 2 above)
Comment (AgentEnder) [reply OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs]: See above

File: OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs (hunk 3 above)
Comment (AgentEnder) [reply OtripleS.Web.Api.Tests.Unit/Services/StudentGuardianServiceTests/StudentGuardianServiceTests.Validations.RetrieveById.cs]: See above

File: OtripleS.Web.Api/Services/StudentGuardians/StudentGuardianService.cs (hunk 4 above)
Comment (AgentEnder) [reply OtripleS.Web.Api/Services/StudentGuardians/StudentGuardianService.cs]: Fixed in latest commit

**Comment by AgentEnder** [comment]
@eriadhami ready for re-review

**Review by eriadhami — APPROVED** [review]
@AgentEnder many thanks

**Comment by AgentEnder** [comment]
@hassanhabib Should this be merged now?

**Comment by hassanhabib** [comment]
@AgentEnder merged.

**Pull request merged**